examples/json-access/json_access.bal

```diff
@@ -0,0 +1,52 @@
+import ballerina/io;
+
+public function main() {
+    // Define a variable of type `json` that holds a mapping value.
+    json j1 = {
+        fname: "Mary",
+        lname: "Jane",
+        address: {
+            line: "20 Palm Grove",
+            city: "Colombo 03",
+            country: "Sri Lanka"
+        }
+    };
+
+    // Field access is allowed on the `json` typed variable, but the return
+    // type would be a union of `json` and `error`. An error will be returned
+    // either if the value on which the access is done is not a mapping or if
+    // it is a mapping, but it does not contain the particular key.
+    json|error r1 = j1.fname;
+    // `r1` is of type `json` since `j1` is a mapping and contains the field `fname`.
+    io:println(r1);
+
+    // Similarly, chained field access is also allowed for `json`. If an `error`
+    // is encountered at any point during the evaluation, evaluation stops at
+    // that point and the result would be an error.
+    json|error r2 = j1.address.city;
+    // `r2` is of type `json` since `j1` is a mapping with the field `address`, which in
+    // turn is a mapping with the field `city` which will be the resultant value.
+    io:println(r2);
+
+    // Attempting to access a non-existent field would result in an error.
+    json|error r3 = j1.age.year;
+    // `r3` is of type `error` since `j1` does not have a field with the key `age`.
```

Comment:
`r3` is of the type `error` since `j1` does not have a field with the key `age`.